src/CMakeLists.txt

```diff
@@ -1,85 +1,85 @@
-            if (HAS_MARCH_NATIVE)
-                target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC -march=native)
-            endif ()
-        endif ()
-    endif ()
+	# set compiler flags depending on compiler
+	if (MSVC)
+		target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC /utf-8)
+	else ()
+		target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC -Wall -Wextra $<$<CONFIG:DEBUG>:-Og>)
+		if (BINDINGS)
+			# adjust visibility settings for building Python bindings
+			target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC -fvisibility=hidden)
+		endif ()
+		if (NOT DEPLOY)
+			# only include machine-specific optimizations when building for the host machine
+			target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC -mtune=native)
+			include(CheckCXXCompilerFlag)
+			check_cxx_compiler_flag(-march=native HAS_MARCH_NATIVE)
+			if (HAS_MARCH_NATIVE)
+				target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC -march=native)
+			endif ()
+		endif ()
+	endif ()
 
-    if (GENERATE_POSITION_INDEPENDENT_CODE OR BINDINGS)
-        include(CheckPIESupported)
-        check_pie_supported()
-        set_target_properties(${PROJECT_NAME}_${LIBNAME}_lib PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
-    endif ()
+	if (GENERATE_POSITION_INDEPENDENT_CODE OR BINDINGS)
+		include(CheckPIESupported)
+		check_pie_supported()
+		set_target_properties(${PROJECT_NAME}_${LIBNAME}_lib PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
+	endif ()
 
-    # add coverage compiler and linker flag to the library and all targets that link against it, if COVERAGE is set
-    if (COVERAGE)
-        target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC --coverage)
-        target_link_libraries(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC --coverage)
-    endif ()
+	# add coverage compiler and linker flag to the library and all targets that link against it, if COVERAGE is set
+	if (COVERAGE)
+		target_compile_options(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC --coverage)
+		target_link_libraries(${PROJECT_NAME}_${LIBNAME}_lib PUBLIC --coverage)
+	endif ()
 
-    # add MQT alias
-    add_library(MQT::${PROJECT_NAME}_${LIBNAME}_lib ALIAS ${PROJECT_NAME}_${LIBNAME}_lib)
+	# add MQT alias
+	add_library(MQT::${PROJECT_NAME}_${LIBNAME}_lib ALIAS ${PROJECT_NAME}_${LIBNAME}_lib)
 endmacro()
 
 if (Z3_FOUND)
-    # exact mapper project library
-    add_qmap_library(exact ExactMapper)
-    # add encodings
-    target_sources(${PROJECT_NAME}_exact_lib PUBLIC
-            ${CMAKE_CURRENT_SOURCE_DIR}/Encodings.cpp
-            ${${PROJECT_NAME}_SOURCE_DIR}/include/Encodings.hpp)
-    # add Z3 library
-    target_link_libraries(${PROJECT_NAME}_exact_lib PUBLIC z3::z3lib)
+	# exact mapper project library
+	add_qmap_library(exact ExactMapper)
+	# add Z3 library
+	target_link_libraries(${PROJECT_NAME}_exact_lib PUBLIC z3::z3lib)
+	# add LogicBlocks library
+	target_link_libraries(${PROJECT_NAME}_exact_lib PUBLIC logicblocks::Logic logicblocks::Encodings)
```

Comment:
It should no longer be necessary to link to Z3 here and to have the inclusion of the Exact Mapper conditioned on `Z3_FOUND` (which shouldn't exist as by the comment above).
It should be enough to link to the LogicBlocks module here and the LogicBlocks library should handle the availability of Z3 and either fail if it is not available or provide a fallback.